packages/components/src/internal/components/search/EntityFieldFilterModal.tsx

```diff
@@ -179,9 +190,13 @@ export const EntityFieldFilterModal: FC<Props> = memo(props => {
             queryLabels[parentValue] = label;
         });
         const filterErrors = getFieldFiltersValidationResult(validDataTypeFilters, queryLabels);
-        if (!filterErrors) onFind(entityDataType.instanceSchemaName, validDataTypeFilters);
-        else setFilterError(filterErrors);
-    }, [onFind, validDataTypeFilters]);
+        if (!filterErrors) {
+            if (metricFeatureArea) api.query.incrementClientSideMetricCount(metricFeatureArea, 'filterModalApply');
+            onFind(entityDataType.instanceSchemaName, validDataTypeFilters);
+        } else {
+            setFilterError(filterErrors);
```

Comment:
Would it be useful to track how often a user generates a bad filter, maybe a sign that our UX needs more polish?

Reply:
Simple enough to track. Added.